eos-phone-home

```diff
@@ -148,6 +153,43 @@ class PhoneHome(object):
 
         return None
 
+    def _get_metrics(self):
+        permissions_file = '/etc/metrics/eos-metrics-permissions.conf'
+        valid_environments = ['production', 'dev', 'test']
+
+        enabled = False
+        uploading_enabled = True
+        environment = 'production'
+
+        try:
+            config = configparser.ConfigParser()
+            config.read(permissions_file)
+            enabled = config['global'].getboolean('enabled', enabled)
+            uploading_enabled = config['global'].getboolean('uploading_enabled',
+                uploading_enabled)
+            environment = config['global'].get('environment', environment)
+        except Exception as e:
+            print("Unable to read %s, assuming fallback values." %
+                (permissions_file, ), e, file=sys.stderr)
+
+        # the event recorder replaces invalid environments with 'test'
+        if environment not in valid_environments:
+            print('Found invalid metrics environment', environment,
+                file=sys.stderr)
+            environment = 'test'
+
+        return (enabled and uploading_enabled, environment)
+
+    def _get_metrics_enabled(self):
+        enabled = self._lookup_or_get_variable('metrics')[0]
+        print(' - Metrics enabled:', enabled)
+        return enabled
+
+    def _get_metrics_environment(self):
+        environment = self._lookup_or_get_variable('metrics')[1]
```

Comment:
This is because we're caching two things in the variable named 'metrics' so we only parse the file once. It's analagous to how _get_live and _get_dualboot work, caching 'cmdline' although it's not sent to the server in the end (although I note _get_dt_info could usefully be called in the same way). I could return a dictionary as metrics if that would make you happier, unless you have a different refactor you'd like to see?